recipe/meta.yaml

```diff
@@ -1,15 +1,14 @@
-{% set version = "3.20.3" %}
+{% set version = "4.23.4" %}
 
 package:
   name: protobuf
   version: {{ version }}
 
 source:
   url: https://github.com/protocolbuffers/protobuf/archive/v{{ version }}/protobuf-v{{ version }}.tar.gz
-  sha256: 9c0fd39c7a08dff543c643f0f4baf081988129a411b977a07c46221793605638
-  patches:  # [py==311]
-    # see https://github.com/protocolbuffers/protobuf/pull/10403
-    - patches/0001-fix-build-with-python-311.patch  # [py==311]
+  sha256: 5f3cd52d6e5062071d99da57a96ea87e39bc020d6d25748001d919c474a4d8ed
+  patches:
+    - patches/0004-use-C-17-everywhere.patch
 
 build:
   number: 0
```

Comment:
Should we update build.sh/bld.bat to run:
`pip install . -vv --install-option="--cpp_implementation" -- no-deps --no-build isolation` like conda-forge did?
See https://github.com/conda-forge/protobuf-feedstock/blob/main/recipe/meta.yaml#L30C23-L30C80

And I wonder if the env variables are needed too https://github.com/conda-forge/protobuf-feedstock/blob/20df32ca656756cbb2401ec50dcbd70ee2d0fe6b/recipe/meta.yaml#L25-L29

Reply:
pip install done, about env variables 2 of them are for cross compilation, the last one for windows and they are in the build section so if build passes I suppose we do need them, I would prefer to not adding them until we know what they for.

Reply:
conda-forge logs say:
`DEPRECATION: --install-option is deprecated because it forces pip to use the 'setup.py install' command which is itself deprecated. pip 23.1 will enforce this behaviour change. A possible replacement is to use --config-settings. Discussion can be found at https://github.com/pypa/pip/issues/11358`
See https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=760479&view=logs&j=d0d954b5-f111-5dc4-4d76-03b6c9d0cf7e&t=841356e0-85bb-57d8-dbbc-852e683d1642&l=214

Reply:
running this on windows, gave error:
`DEPRECATION: protobuf is being installed using the legacy 'setup.py install' method, because the '--no-binary' option was enabled for it and this currently disables local wheel building for projects that don't have a 'pyproject.toml' file. pip 23.1 will enforce this behaviour change. A possible replacement is to enable the '--use-pep517' option. Discussion can be found at https://github.com/pypa/pip/issues/11451`